projects/applicationinsights-angularplugin-js/src/lib/applicationinsights-angularplugin-js.component.ts

```diff
@@ -35,50 +37,78 @@ export class AngularPlugin extends BaseTelemetryPlugin {
     public priority = 186;
     public identifier = 'AngularPlugin';
 
-    private analyticsPlugin: IAppInsights;
+    private analyticsPlugin: AnalyticsPlugin;
     private propertiesPlugin: properties.PropertiesPlugin;
+    private _angularCfg: IAngularExtensionConfig = null;
+    private _eventSubscription: Subscription = null;
+
+
+    constructor() {
+        super();
+        let self = this;
+
+        self._doTeardown = (unloadCtx?: IProcessTelemetryUnloadContext, unloadState?: ITelemetryUnloadState, asyncCallback?: () => void): void | boolean => {
+            if (this.analyticsPlugin && ApplicationinsightsAngularpluginErrorService.instance !== null) {
+                ApplicationinsightsAngularpluginErrorService.instance.plugin = null;
+                if (self._angularCfg) {
+                    if (self._angularCfg.errorServices && Array.isArray(self._angularCfg.errorServices)) {
+                        arrForEach(self._angularCfg.errorServices, (errorService: IErrorService) => {
+                            ApplicationinsightsAngularpluginErrorService.instance.removeErrorHandler(errorService);
+                        });
+                    }
+                }
+            }
+
+            if (self._eventSubscription) {
+                self._eventSubscription.unsubscribe();
+                self._eventSubscription = null;
+            }
+
+            self.analyticsPlugin = null;
+            self.propertiesPlugin = null;
+            self._angularCfg = null;
+        }
+    }
 
     initialize(config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) {
         super.initialize(config, core, extensions, pluginChain);
         const self = this;
         const ctx = self._getTelCtx();
         const extConfig = ctx.getExtCfg<IAngularExtensionConfig>(self.identifier, { router: null, errorServices: null });
-        arrForEach(extensions, ext => {
-            const identifier = (ext as ITelemetryPlugin).identifier;
-            if (identifier === 'ApplicationInsightsAnalytics') {
-                self.analyticsPlugin = (ext as any) as IAppInsights;
-                if (ApplicationinsightsAngularpluginErrorService.instance !== null) {
-                    ApplicationinsightsAngularpluginErrorService.instance.plugin = self.analyticsPlugin;
-                    if (extConfig.errorServices && Array.isArray(extConfig.errorServices)) {
-                        arrForEach(extConfig.errorServices, (errorService: IErrorService) => {
-                            ApplicationinsightsAngularpluginErrorService.instance.addErrorHandler(errorService);
-                        });
-                    }
+
+        self.propertiesPlugin = core.getPlugin<properties.PropertiesPlugin>(PropertiesPluginIdentifier)?.plugin;
```

Comment:
Using the new `getPlugin` API to get the instance, rather than the previous array iteration approach.